apiml-common/src/main/java/com/ca/mfaas/product/logging/ApimlLogInjector.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.product.logging;
+
+import com.ca.mfaas.message.core.MessageService;
+import com.ca.mfaas.message.log.ApimlLogger;
+import com.ca.mfaas.product.logging.annotations.InjectApimlLogger;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ReflectionUtils;
+
+/**
+ * This class plays as a injector for {@link ApimlLogger} in spring environment.
+ * It detects if class have field with {@link InjectApimlLogger} annotation,
+ * it finds {@link MessageService} from Spring context then initialize {@link ApimlLogger}
```

Comment:
Can you be more specific here? Does the '{@link MessageService}' triggers the "initialization" (whatever that means) of the '{@link ApimlLogger}' if found in Spring context? What if it's not found?

Reply:
The MessageService is used for the initialisation of ApimlLogger. So the ApimlLogInjector detects if a generic class A has a Spring annotation @InjectApimlLogger and if it has, it inject the ApimLogger bean into this class which is initialised with the MessageService (which is passaged as a parameter to the ApimlLogger).

Reply:
Then "If {@link ApimlLogger} detects that the generic class A has a {@InjectApimlLogger} Spring annotation, {@link ApimlLogger} will be initialized with the {@link MessageService}." 
BUT! Tbh, I'm unsure, if it's incorrect, let's discuss in person tomorrow. I'm going to stop by in your office.